From 8954db44d8226540957359fc676f04c26dcdfd9e Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Sat, 20 Feb 2016 08:56:23 +0000
Subject: [PATCH] search/lib.c: cleanup -Wcast-qual warning

lib.c:117:17: warning: cast discards 'const' qualifier from pointer target type [-Wcast-qual]
     next_char = (char *) str_cget_next_char (converted_str);
                 ^

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 lib/search/lib.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/search/lib.c b/lib/search/lib.c
index cc60ff4..5d086f1 100644
--- a/lib/search/lib.c
+++ b/lib/search/lib.c
@@ -97,7 +97,8 @@ gchar *
 mc_search__get_one_symbol (const char *charset, const char *str, gsize str_len,
                            gboolean * just_letters)
 {
-    gchar *converted_str, *next_char;
+    gchar *converted_str;
+    const gchar *next_char;
 
     gsize tmp_len;
 #ifdef HAVE_CHARSET
@@ -114,7 +115,7 @@ mc_search__get_one_symbol (const char *charset, const char *str, gsize str_len,
     converted_str = g_strndup (str, str_len);
 #endif
 
-    next_char = (char *) str_cget_next_char (converted_str);
+    next_char = str_cget_next_char (converted_str);
 
     tmp_len = next_char - converted_str;
 
